test/format/config.c

```diff
@@ -275,6 +278,22 @@ config_backup(void)
         config_single(cstr, false);
     }
 }
+
+/*
+ * config_backup --
+ *     Backup configuration.
+ */
+static void
+config_backward_compatible(void)
+{
+    if (!g.backward_compatible)
+        return;
+
+    if (config_is_perm("disk.mmap_all"))
+        testutil_die(EINVAL, "-B option incompatible with mmap_all configuration");
```

Comment:
For my own understanding, what is `mmap_all` and why it's incompatible?

Reply:
It's a feature recently added that does both reads & writes via mmap (the `mmap` configuration does reads via mmap, the `mmap_all` does reads & writes via mmap).

It's incompatible because 4.2 releases can't handle that configuration string.

This is actually a potentially interesting question: if a MongoDB installation configures a database with a WiredTiger option that's not supported in previous releases, then they've broken their path to downgrading.

Reply:
Thanks for the explanation. 
I'm not sure if the equivalents wiredtiger config options for format`mmap` & `mmap_all` are exposed to MongoDB users, but what you pointed out sounds like a legit concern in general. Maybe the MongoDB FCV mechanism can be leveraged to instruct users to disable those  WiredTiger config options newly introduced in the release before downgrading.

Reply:
It's actually not enough to disable config options -- if it's a per-file option (for example, "mmap_all"), it's stored in the metadata for the file, and any attempt to open the file in a previous release will fail.

Reply:
I assume that is the only new config option since 4.2?

Reply:
Is that true? Maybe I'm getting confused in this conversation about various configurations (format config file, config strings passed to WT, metadata config strings stored on disk, etc). But I thought that older versions can handle a superset of metadata. That is, as long as whatever 4.2 is expecting can be parsed out of the metadata string it reads, it is happy. In 4.4 we've also added the `checkpoint_backup_info` to the metadata string.

Reply:
You're right, I'm not being precise (and I'm honestly not sure of the exact parameters). I'm sure that additional strings in the metadata are fine (and we've added stuff in there quite a bit). But, I believe that if you add stuff to a file's configuration, or to the base configuration, then things will fail.